dbt/adapters/spark/connections.py

```diff
@@ -88,10 +91,25 @@ def cursor(self):
 
     def cancel(self):
         if self._cursor is not None:
-            self._cursor.cancel()
+            # Handle bad response in the pyhive lib when
+            # the connection is cancelled
+            try:
+                self._cursor.cancel()
+            except OSError as exc:
```

Comment:
You might want to make these into `EnvironmentError`s unless you're totally sure it'll only raise `OSError`s. In python 2.7, a lot of networking stuff raises weird exceptions that descend from `EnvironmentError` but not `OSError` in certain edge cases (in 3.x they are the same)

Reply:
I didn't know about `EnvironmentError`! I just threw these in because cancellation resulted in a pretty gnarly (and unhelpful) stacktrace. Updating!